lib/geohash.ex

```diff
@@ -110,6 +110,83 @@ defmodule Geohash do
     {lat, lon}
   end
 
+  @doc ~S"""
+  Calculate adjacent/2 geohash in ordinal direction ["n","s","e","w"]
+  Deals with boundary cases when adjacent is not of the same prefix.
+
+  ##Examples
+  ```
+  iex> Geohash.adjacent("abx1","n")
+  "abx4"
+
+```
+  """
+  def adjacent("",_direction) do
+    {:error, "empty geohash"}
+  end
+  def adjacent(geohash,direction) when is_list(geohash) do
+    adjacent(to_string(geohash),direction)
+  end
+  def adjacent(geohash,direction) when is_bitstring(geohash) do
+    adjacent_aux(String.downcase(geohash),String.downcase(to_string(direction)))
+  end
+  @neighbor %{
+    "n" => { 'p0r21436x8zb9dcf5h7kjnmqesgutwvy', 'bc01fg45238967deuvhjyznpkmstqrwx' },
+    "s" => { '14365h7k9dcfesgujnmqp0r2twvyx8zb', '238967debc01fg45kmstqrwxuvhjyznp' },
+    "e" => { 'bc01fg45238967deuvhjyznpkmstqrwx', 'p0r21436x8zb9dcf5h7kjnmqesgutwvy' },
+    "w" => { '238967debc01fg45kmstqrwxuvhjyznp', '14365h7k9dcfesgujnmqp0r2twvyx8zb' },
+  }
+  @border %{
+    "n" => { 'prxz',     'bcfguvyz' },
+    "s" => { '028b',     '0145hjnp' },
+    "e" => { 'bcfguvyz', 'prxz'     },
+    "w" => { '0145hjnp', '028b'     },
+  }
+
+  defp border_case(direction,type,tail) do
+    elem(@border[direction],type) |>
+    Enum.find_index(fn r -> r==tail end)
+  end
+  defp adjacent_aux(geohash,direction) when direction in ["n","s","w","e"] do
+    prefix_len = byte_size(geohash)-1
+    # parent will be a string of the prefix, lastCh will be an int of last char
+    <<parent::binary-size(prefix_len), lastCh::size(8)>> = geohash
+    type = rem(prefix_len+1,2)
+ 
+    # check for edge-cases which don't share common prefix
+    parent = if ( border_case(direction,type,lastCh) && prefix_len > 0 ) do
+      adjacent_aux(parent,direction)
+    else
+      parent
+    end
+
+    # append letter for direction to parent
+    # look up index of last char use as position in base32
+    pos = @neighbor[direction] |>
+      elem(type) |>
+      Enum.find_index(fn r -> r==lastCh end)
+
+    q = Enum.slice(@geobase32, pos, 1)
+    parent <> to_string(q)
+  end
+  @doc ~S"""
+  Calculate adjacent hashes for the 8 touching neighbors/1
+
+  ## Examples
+
+  ```
+  iex> Geohash.neighbors("abx1")
+  [{"n", "abx4"}, {"s", "abx0"}, {"e", "abx3"}, {"w", "abwc"}, {"ne", "abx6"},
+ {"se", "abx2"}, {"nw", "abwf"}, {"sw", "abwb"}]
+
+  ```
+  """
+  def neighbors(geohash) do
```

Comment:
@danj3 I was reading the project you used as a base and I find their implementation easier to understand:
https://github.com/chrisveness/latlon-geohash/blob/master/latlon-geohash.js#L229

Also, if this function returns a map like `%{"n" => "abx4", "s" => "abx0", ...}` I could do things like:
```
iex> %{"s" => s, "e" => e} = Geohash.neighbors("abx1")
iex> s
"abx0"
iex> e
"abx3"
```

What do you think?